apps/web/lib/hasEmptyAssignment.ts

```diff
@@ -0,0 +1,43 @@
+import type { SchedulingType } from "@calcom/prisma/enums";
+
+import type { EventTypeAssignedUsers, EventTypeHosts } from "~/event-types/views/event-types-single-view";
+
+// This function checks if EventType requires assignment.
+// returns true: if EventType requires assignment but there is no assignment yet done by the user.
+// returns false: for all other scenarios.
+export function hasEmptyAssignment({
+  assignedUsers,
+  hosts,
+  schedulingType,
+  assignAllTeamMembers,
+  isTeamEvent,
+}: {
+  assignedUsers: EventTypeAssignedUsers;
+  hosts: EventTypeHosts;
+  schedulingType: SchedulingType | null;
+  assignAllTeamMembers: boolean;
+  isTeamEvent: boolean;
+}): boolean {
+  // if Non-Team event, no need to check for assignments.
+  if (!isTeamEvent) {
+    return false;
+  }
```

Comment:
@vijayraghav-io I suggested you make this independent test in the `handleRouteChange` function not here. Should be present above this `!leaveWithoutAssigningHosts.current &&`